### PR TITLE
Add backup service descriptor to service instance spec

### LINKED ISSUE #4: Add backup service descriptor
Add backup service descriptor to service instance struct.

### PR DESCRIPTION
**Pull request by sancyx**
| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | yes
| API breaks?     | no
| Deprecations?   | no
| Related tickets | fixes #4 
| License         | Apache 2.0

### What's in this PR?
Add backup service descriptor to service instance spec.

### Checklist

- [x] Code meets the [Developer Guide](https://github.com/banzaicloud/developer-guide)
- [x] User guide and development docs updated (if needed)
- [x] Related Helm chart(s) updated (if needed)

### CODE REVIEW

**Review by lpuskas — APPROVED** [review]
lgtm

File: api/v1alpha1/backup/backup.go [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+	"emperror.dev/errors"
+	"github.com/mitchellh/mapstructure"
+	v1 "k8s.io/api/core/v1"
+)
+
+type ServiceSpec struct {
+	ChartValues ValueOverrides `json:"chartValues"`
+}
+
+type ValueOverrides struct {
```
Comment (pepov) [inline api/v1alpha1/backup/backup.go:31]: All these configuration are required? I guess some of these could be `omitempty`

Comment (sancyx) [reply api/v1alpha1/backup/backup.go:37]: yes, I've added `omitempty` on some of them.

**Review by pregnor — COMMENTED** [review]
LGTM, although I don't have much context about the high level expectations of an integrated service.

**Review by pepov — APPROVED** [review]

**Pull request merged**